fix: web form not submitting data in grid views

fumin opened the pull request:
Web forms work nicely with vanilla type fields. However, for child tables fields in grid views, the data is not submitted.
The below instructions show how to replicate this issue:

Create a doctype with a child table field. In this case the doctype is "coachForm" the field is "指導選手參賽紀錄"
![Screen Shot 2020-04-05 at 3 42 39 AM](https://user-images.githubusercontent.com/765222/78460160-2220eb00-76f1-11ea-8665-7a2a74e28ebc.png)

The "指導選手參賽紀錄" child table field maps to the "coachForm_medal" doctype
![Screen Shot 2020-04-05 at 3 42 48 AM](https://user-images.githubusercontent.com/765222/78460191-57c5d400-76f1-11ea-833f-038a7e357f2c.png)

Create a web form for the "coachForm" doctype and submit it.
Note that the variable `this.doc` does not contain the info relating to the child table "指導選手參賽紀錄" entered by the user.
![Screen Shot 2020-04-05 at 3 44 22 AM](https://user-images.githubusercontent.com/765222/78460242-c440d300-76f1-11ea-9832-730501745794.png)

Since the web form did not submit data for the mandatory field "指導選手參賽紀錄", an error is returned.
![Screen Shot 2020-04-05 at 3 44 39 AM](https://user-images.githubusercontent.com/765222/78460249-dcb0ed80-76f1-11ea-9935-7eaf0040acbd.png)

This might be related to https://github.com/frappe/frappe/issues/8932

fumin committed:
fix: web form not submitting data in grid views

Web forms work nicely with vanilla type fields. However, for child tables fields in grid views, the data is not submitted.
The below instructions show how to replicate this issue:

fumin commented:
@scmmishra Hi, I wonder if it's possible for you to have a look at this pull request? Our site relies on this ability to submit forms. Thanks!

scmmishra commented:
> @scmmishra Hi, I wonder if it's possible for you to have a look at this pull request? Our site relies on this ability to submit forms. Thanks!

Yeah, I plan on either merging this or fixing it in a seperate PR by the end of week.

fumin commented:
thanks @scmmishra happy to provide more information anytime

scmmishra committed:
fix: update doc object with current values

scmmishra commented:
@fumin I've made a few changes to the PR and approved it, it'll be merged automatically when the tests pass

fumin commented:
@scmmishra thanks for the changes, they look good.
The build seems to fail on

```
Timed out retrying: expected '#List/Website/List' to equal '#List/Contact/List'
```

which seems unrelated to the change here. Given the build succeeded previously, perhaps this is a flaky test?

scmmishra commented:
> which seems unrelated to the change here. Given the build succeeded previously, perhaps this is a flaky test?

Yeah, the UI tests are not very predictable, there are a few issues which were fixing. I'll restart the test.

fumin commented:
@scmmishra thanks for restarting the test which seemed to pass this time.
Could you help us start the Codacy build, too? Thanks

fumin committed:
add semi-colon to end of line

Codacy demands that we add this semi-colon https://app.codacy.com/gh/frappe/frappe/pullRequest?prid=5305922 .

scmmishra commented:
@Mergifyio backport version-12-hotfix

fumin commented:
@scmmishra after fixing a semi-colon issue raised by Codacy, the Travis build seems to fail on the same flaky test "forms.js", I wonder could you help me restart the build?
Also, I seem not to be able to view the security/snyk page with error "Unable to display this organization". I wonder could you help me look into it?
Thanks

![Screen Shot 2020-04-11 at 5 03 04 PM](https://user-images.githubusercontent.com/765222/79039809-6a229f00-7c16-11ea-947f-6134a05280a3.png)

scmmishra committed:
Merge branch 'develop' of github.com:frappe/frappe into patch-1

scmmishra commented:
@fumin I've updated the branch, the synk issue will be fixed now. The tests will pass soon and we'll merge this

fumin commented:
@scmmishra Thank you for your help : )

dinukasal commented:
Thank you guys! @fumin  @scmmishra

shubhamgupta123 commented:
@Mergifyio backport version-12-hotfix
